src/MongoDB/Driver/Manager.cpp

```diff
@@ -238,34 +238,36 @@ static bool hippo_mongo_driver_manager_apply_rp(mongoc_uri_t *uri, const Array o
 		mongoc_read_prefs_set_tags(new_rp, b_tags);
 	}
 
-	/* Handle maxStalenessMS, and make sure it is not combined with PRIMARY readPreference */
+	/* Handle maxStalenessSeconds, and make sure it is not combined with PRIMARY readPreference */
 	if (
-		(options.exists(s_MongoDBDriverManager_maxstalenessms) && options[s_MongoDBDriverManager_maxstalenessms].isInteger())
+		(options.exists(s_MongoDBDriverManager_maxstalenessseconds) && options[s_MongoDBDriverManager_maxstalenessseconds].isInteger())
 		||
-		(options.exists(s_MongoDBDriverManager_maxStalenessMS) && options[s_MongoDBDriverManager_maxStalenessMS].isInteger())
+		(options.exists(s_MongoDBDriverManager_maxStalenessSeconds) && options[s_MongoDBDriverManager_maxStalenessSeconds].isInteger())
 	) {
-		int64_t maxStalenessMS = 0;
+		int64_t maxStalenessSeconds = 0;
 
 		if (mongoc_read_prefs_get_mode(new_rp) == MONGOC_READ_PRIMARY) {
-			throw MongoDriver::Utils::throwInvalidArgumentException("Primary read preference mode conflicts with maxStalenessMS");
+			throw MongoDriver::Utils::throwInvalidArgumentException("Primary read preference mode conflicts with maxStalenessSeconds");
```

Comment:
The conflict with a primary read preference mode should only be thrown if the user sets `maxStalenessSeconds` to a positive value. Let's move this after the range checking and only check if the assigned value is not `MONGOC_NO_MAX_STALENESS`.